ValveKeyValue/ValveKeyValue/Serialization/KV1BinarySerializer.cs

```diff
@@ -78,13 +78,12 @@ void Write(KV1BinaryNodeType nodeType)
             writer.Write((byte)nodeType);
         }
 
-        void Write(string value)
+        void Write(string value, Encoding encoding = null)
         {
-            foreach (var @char in value)
-            {
-                writer.Write(@char);
-            }
+            if (encoding == null) encoding = Encoding.ASCII;
```

Comment:
As above regarding NUL termination, but we don't really have a better option here.

Reply:
Modified to write a terminated byte array instead of a string and moved encoding to the caller.